tests/integration/cairo_path/cairo_path_test.py

```diff
@@ -0,0 +1,103 @@
+from tests.integration.cairo_compiler.prepare_files_fixture import (
+    PrepareFilesFixture,
+    RequestedFile,
+)
+
+
+@pytest.fixture(name="prepare_files")
+def prepare_files_fixture(create_protostar_project: CreateProtostarProjectFixture):
+    with create_protostar_project() as protostar:
+        protostar.build_sync()
+        return PrepareFilesFixture(protostar)
+
+
+SCRIPT_ROOT = Path(__file__).parent
+CONTRACTS_DIR = SCRIPT_ROOT / "contracts"
+
+
+def test_cairo_path_for_starknet_contract():
+    casm_contents = call_starknet_contract_compiler(
+        input_path=CONTRACTS_DIR / "starknet_project" / "basic_starknet_contract.cairo",
+        cairo_paths=[CONTRACTS_DIR / "external_lib"],
+    )
+    assert casm_contents
+
+    with pytest.raises(Exception):
+        call_starknet_contract_compiler(
+            input_path=CONTRACTS_DIR
+            / "starknet_project"
+            / "basic_starknet_contract.cairo",
+        )
+
+
+def test_cairo_path_for_starknet_test(prepare_files: PrepareFilesFixture):
+    prepared_files = prepare_files.prepare_files(
+        requested_files=[
+            RequestedFile.output_sierra,
+        ]
+    )
+
+    with pytest.raises(Exception):
+        call_test_collector(input_path=CONTRACTS_DIR / "starknet_project")
+
+    _, test_names = call_test_collector(
+        input_path=CONTRACTS_DIR / "starknet_project",
+        output_path=prepared_files["output_sierra"][0],
+        cairo_paths=[CONTRACTS_DIR / "external_lib"],
+    )
+    assert test_names
+    protostar_casm = call_protostar_sierra_to_casm(
+        named_tests=test_names,
+        input_path=prepared_files["output_sierra"][0],
+    )
+    assert protostar_casm
+
+
+def test_cairo_path_for_regular_compiler(prepare_files: PrepareFilesFixture):
+    prepared_files = prepare_files.prepare_files(
+        requested_files=[
+            RequestedFile.output_sierra,
+            RequestedFile.output_casm,
+        ]
+    )
+
+    # cairo -> sierra -> casm
+    with pytest.raises(Exception):
+        call_cairo_to_sierra_compiler(
+            input_path=CONTRACTS_DIR / "regular_project" / "mycontract.cairo"
+        )
+
+    call_cairo_to_sierra_compiler(
+        input_path=CONTRACTS_DIR / "regular_project" / "mycontract.cairo",
+        output_path=prepared_files["output_sierra"][0],
+        cairo_paths=[CONTRACTS_DIR / "external_lib"],
+    )
+    casm_contents = call_sierra_to_casm_compiler(
+        input_path=prepared_files["output_sierra"][0],
+    )
+    assert casm_contents
+
+    # cairo -> casm
+    with pytest.raises(Exception):
+        call_cairo_to_casm_compiler(
+            input_path=CONTRACTS_DIR / "regular_project" / "mycontract.cairo"
+        )
+
+    casm_contents = call_cairo_to_casm_compiler(
+        input_path=CONTRACTS_DIR / "regular_project" / "mycontract.cairo",
+        cairo_paths=[CONTRACTS_DIR / "external_lib"],
+    )
+    assert casm_contents
```

Comment:
This could be a separate test case.

Reply:
The tests for functions `call_cairo_to_sierra_compiler`, `call_sierra_to_casm_compiler`, and `call_cairo_to_casm_compiler` are in one test case in `simple_contracts_compiler_test.py` merged in #1316. We should have a consistent design in both places and changing the code in `simple_contracts_compiler_test.py` in this PR is beyond of its scope. I consider this as a nitpick, however, if you feel strongly about this, you can spawn an issue for this.